### ISSUE TITLE
Apex -- No Module named amp_C

### ISSUE THREAD

**Issue by pradeep-vishnu** [robert80203/GSVNet#2]
Hello Robert,

I'm really impressed by your publication and the benchmark results. Therefore I came here to check it out further. 

I'm facing issues with the apex dependency. I followed the same steps as described and even tried a few others as well, but I'm unable to find a workaround.
I keep getting the below error when I run: `python main.py --evaluate 1 --batch-size 1 --resume 1`

Have you ever encountered a similar issue? If yes, can you please share how you solved the same? 

```
Traceback (most recent call last):
  File "/home/vpradeep/Documents/repos/GSVNet/trainer/base.py", line 15, in <module>
    from apex.parallel import DistributedDataParallel, SyncBatchNorm
  File "/home/vpradeep/anaconda3/envs/gsvnet/lib/python3.7/site-packages/apex/__init__.py", line 20, in <module>
    from . import transformer
  File "/home/vpradeep/anaconda3/envs/gsvnet/lib/python3.7/site-packages/apex/transformer/__init__.py", line 3, in <module>
    from apex.transformer import pipeline_parallel
  File "/home/vpradeep/anaconda3/envs/gsvnet/lib/python3.7/site-packages/apex/transformer/pipeline_parallel/__init__.py", line 1, in <module>
    from apex.transformer.pipeline_parallel.schedules import get_forward_backward_func
  File "/home/vpradeep/anaconda3/envs/gsvnet/lib/python3.7/site-packages/apex/transformer/pipeline_parallel/schedules/__init__.py", line 4, in <module>
    from apex.transformer.pipeline_parallel.utils import get_num_microbatches
  File "/home/vpradeep/anaconda3/envs/gsvnet/lib/python3.7/site-packages/apex/transformer/pipeline_parallel/utils.py", line 23, in <module>
    import amp_C
ModuleNotFoundError: No module named 'amp_C'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "main.py", line 60, in <module>
    from trainer.gsvnet_trainer import GSVNet_Trainer
  File "/home/vpradeep/Documents/repos/GSVNet/trainer/gsvnet_trainer.py", line 12, in <module>
    from trainer.base import Trainer
  File "/home/vpradeep/Documents/repos/GSVNet/trainer/base.py", line 18, in <module>
    "Please install apex from https://www.github.com/nvidia/apex .")
ImportError: Please install apex from https://www.github.com/nvidia/apex .
```

**Comment by pradeep-vishnu** [comment]
I further tried by bypassing this import. As it turns out, the other dependencies you have with the apex package has similar issues. So it would really help if you could specify which stable commit or the version of the apex you have used during your development.

**Comment by robert80203** [comment]
You may follow the instructions in the official github of apex
https://github.com/NVIDIA/apex
There are some issues about your case.
https://github.com/NVIDIA/apex/issues/573
Hope these will help you

**Comment by pradeep-vishnu** [comment]
Thank you for replying. I already saw the issues section in the apex repo. But those suggestions did not work for me. The issue I cited could be resolved by installing the full apex package with the C++ and CUDA extension `git clone https://github.com/NVIDIA/apex`
`cd apex`
`pip install -v --disable-pip-version-check --no-cache-dir --global-option="--cpp_ext" --global-option="--cuda_ext" ./
` instead of installing the python only version of apex, which you have stated.
